### PR TITLE
Adds a code style rule to check if there are unused imports

### PR DESCRIPTION
**Pull request by laoneo**
Adds a new code style rule which checks if there are [unused imports](https://cs.symfony.com/doc/rules/import/no_unused_imports.html). This rule has some caveats with [comments](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/2814), but the rest should work pretty stable.

### CODE REVIEW

File: libraries/src/Document/HtmlDocument.php [hunk 1]
```diff
@@ -18,7 +18,6 @@
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\Uri\Uri;
 use Joomla\CMS\Utility\Utility;
-use Joomla\Database\ParameterType;
```
Comment (brianteeman) [inline libraries/src/Document/HtmlDocument.php:21]: ~This is used on line 711 if i understand correctly~

`                    ->bind(':id', $active->id, ParameterType::INTEGER);
`

Comment (brianteeman) [reply libraries/src/Document/HtmlDocument.php:21]: i might be looking at an old branch on my phone.

Comment (brianteeman) [reply libraries/src/Document/HtmlDocument.php:21]: sorry it was an old branch. all good

**Comment by laoneo** [comment]
Check the touched files if the imports are really not used. Classes like document or the view are used across the whole code base, so are tested with the system tests. The rest can be tested when you actually use their functionality like the SQL custom field or tags.

**Pull request merged**